app/controllers/api/services_controller.rb

```diff
@@ -143,8 +143,28 @@ def queue_chargeback_report_resource(_type, id, _data)
       action_result(false, "Could not queue chargeback report generation - #{err}")
     end
 
+    def retire_resource(type, id = nil, data = nil)
```

Comment:
We should probably fix this in `base_controller/generic.rb` and remove the methods from VmsController if possible.

Reply:
yeaa, it would be more correct, but this PR is very looks very braking for backporting